[Zhao Huan] iP

zhaohuanqdcn opened the pull request.

Commit:
Add Gradle support

zhaohuanqdcn committed:
Level-1

Greet, Echo, Exit are implemented

Greet: auto-triggered upon running
Echo: echo any input from the user
Exit: exit on commands containing "bye" (not case-sensitive)

zhaohuanqdcn committed:
Level-2

Basic task list functions are implemented:

Add: add any input to the task list
List: display all tasks stored

zhaohuanqdcn committed:
Level-3

A basic tesk list function is implemented:

Done: mark a task as done using commands "done <index>"

Tasks are represented in a separate class (as in level-2)

zhaohuanqdcn committed:
Level-4

New task list features are implemented:

Todo: normal task inputs
Deadline: commands of format "deadline <some task> /by <some time>"
Event: commands of format "event <some task> /at <some time>"

The task class is now abstract, with Todo & Deadline & Event extending it

zhaohuanqdcn committed:
Level-4 Fix

Fix the Todo feature:

Todo: commands in format "todo <some task>" without time attached

zhaohuanqdcn committed:
A-TextUiTesting

zhaohuanqdcn committed:
Level-5

Error handling features are implemented:

Todo without a task, event/deadline without a task or a time, or any invalid input format are handled as exceptions

zhaohuanqdcn committed:
Level-6

A basic task list feature is implemented:

Delete: delete a task from the list using its index

Error handling: handled exceptions caused by invalid indices from done and delete

zhaohuanqdcn committed:
A-Enums

Task types are updated as enum to improve robustness

zhaohuanqdcn committed:
Update save feature

Update save feature for task lists

Data can not be saved and Duke is reloaded after restart.

Now task list related funtions (including list, done, add, delete) can have stored data so that after restart Duke can still access the previous task list in the file.

zhaohuanqdcn committed:
Update date support

Add date support for event / deadline

Event and deadline take in the non-empty string after the key word ("/at" and "/by" respectively) as a time and store it

Now event and deadline have date support if the commands are in following formats:
 - event <task> /at yyyy-MM-dd
 - deadline <task> /by yyyy-MM-dd
These commands are still compatible with other types of inputs and just treat them as a string as before.

zhaohuanqdcn committed:
Merge branch 'branch-level-8'

zhaohuanqdcn committed:
Class Abstraction

Reorganize existing code via classes

Duke is devided into 5 main components:
 - Ui (extracted from Main),
 - Storage (adapted from FileController),
 - Parser (existing class),
 - TaskList (adapted from ListFunction) and
 - Command (extracted from Main).

zhaohuanqdcn committed:
Package the code

Pack the code into 3 packages

Existing classes are packed into 3 packages:
 - duke,
 - task,
 - utility.

zhaohuanqdcn committed:
Add JUnit tests

Add JUnit tests for Parser and Task

Now a JUnit test directory is created with test code located at /src/test/java. Test drivers for Parser and Task are set up.

zhaohuanqdcn committed:
Add JavaDoc

Add JavaDoc support to most methods

zhaohuanqdcn committed:
Update Coding Standard

Reformat code to SE coding standard

zhaohuanqdcn committed:
Add Find Feature

Add find function in task list

Tasks can be viewed on through the "list" command, which can get more difficult to find the relevent task as the list size increases.

Now users can search for a task using a keyword with the help of Find command:
 - find <keyword>
This will show a list of tasks that contains the key word given.

zhaohuanqdcn committed:
Merge branch 'branch-A-CodingStandard'

zhaohuanqdcn committed:
Merge branch 'branch-level-9'

zhaohuanqdcn committed:
Merge Branches

Merge braches to master

zhaohuanqdcn committed:
Merge branch 'add-gradle-support'

zhaohuanqdcn committed:
Add Gradle Support

Add gradle support and update txt ui testing

zhaohuanqdcn committed:
Add GUI Support

Add GUI Support

The previous command line interface in terminal is not user friendly and often faces display issues.

A GUI using JavaFX is added with help of the newly implemented gui package. Entities in duke package are updated to accomodate gui changes.

KishenKumarrrrr commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,80 @@
+import java.security.InvalidParameterException;
+
+public class Command {
+    private final String line;
```
Perhaps the variable could have a slightly more descriptive name?

KishenKumarrrrr commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,192 @@
+        StringBuilder sb = new StringBuilder();
+        sb.append("Here are the matching tasks in your list:");
+        int count = 1;
+        boolean found = false;
```
_found_ sounds alright but I don't know if it is required for it to be named as a question. 🤔

KishenKumarrrrr commented on src/main/java/task/Deadline.java:
```diff
@@ -0,0 +1,30 @@
+import utility.DateParser;
+
+public class Deadline extends Task {
+    private final String ddl;
```
Perhaps we should not abbreviate variable names?

KishenKumarrrrr commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,41 @@
+
+public abstract class Task {
+    protected String name;
+    protected boolean done;
```
Perhaps try to name the boolean variable as a question?

KishenKumarrrrr reviewed (commented):
Good job! I felt that your code was very well written :)

AdithyaNarayan commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,66 @@
+     *
+     * @return an error message
+     */
+    public static String commandError() {
```
A clear idea of method intent is conveyed. However, it would be more coherent with the rest of the names if it was a verb. Maybe something along the lines of `getErrorForCommand`?

AdithyaNarayan commented on src/main/java/gui/Main.java:
```diff
@@ -0,0 +1,33 @@
+/**
+ * A GUI for Duke using FXML.
+ */
+public class Main extends Application {
```
I feel that the class name is a bit ambiguous.

AdithyaNarayan reviewed (commented):
Great job on the iP! I found some useful tips which I can use in my implementation also! Keep up the good work!

zhaohuanqdcn committed:
Improve Code Quality

Improve overall code quality

- Main class is renamed to MainApplication.
- All ui components inside TaskList class are extracted.
- Some variables are renamed.
- Error handling clauses in Command class are refactored.
- Inheritted methods in task package are refactored.

zhaohuanqdcn committed:
Add Assertions

RuiFengg commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,188 @@
+                count++;
+            }
+        }
+        if (!isFound) {
```
The boolean isFound might be redundant since the toString() method for string builder will return an empty String if nothing is appended.

RuiFengg commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,29 @@
+        this.eventTime = eventTime;
+        try {
+            date = LocalDate.parse(eventTime);
+        } catch (Exception e) {
```
Consider catching a more specific exception such as DateTimeParseException that is thrown by the parse method when the text cannot be parsed. Hovering over methods can provide some information on specific exceptions thrown by library methods.

RuiFengg commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,115 @@
+
+import java.security.InvalidParameterException;
+
+public class Parser {
```
Nice implementation for parser, I could learn from this.

RuiFengg commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,181 @@
+package duke;
```
Interesting implementation for storing the tasklist, learnt another way of approaching this from your code. I only stored the task list when the bye command was issued but yours was to store every time commands were entered

RuiFengg reviewed (commented):
Overall neat and clean code, just a few minor nits

zhaohuanqdcn committed:
Improve Code Quality

Fix problems mentioned in PR reviews

zhaohuanqdcn committed:
Update Archive Feature

Add archive feature to duke

Tasks can only be deleted forever if a cleaner task list is desired.

Now archive feature is added, so that users can wipe out tasks from the main list and can still access them in the archived list if needed.
New command:
- list archived: lists all tasks from the archived task list
- archive <index>: archives the task of given index from the main list

zhaohuanqdcn committed:
Update to Better GUI

Update to better GUI

Newly add clipped avatars, bubbled speeches and a consistent icon for GUI panel

zhaohuanqdcn committed:
Update User Guide

zhaohuanqdcn committed:
Set theme jekyll-theme-slate

zhaohuanqdcn committed:
Fix UG Typo

zhaohuanqdcn committed:
Merge branch 'master' of https://github.com/zhaohuanqdcn/ip

zhaohuanqdcn committed:
Revert "Merge branch 'master' of https://github.com/zhaohuanqdcn/ip"

This reverts commit 66f7bb1782baffd18fb8d5347f7df9ba7737c8e9, reversing
changes made to ff7ae01d737b1b25682449a5068ee7154a23b2c9.

zhaohuanqdcn committed:
Fix Readme and Filepaths

Update for release

- Change filepaths to relative paths.
- Fix typo in the userguide.
